pkg/provisioner/ironic/clients/client.go

```diff
@@ -46,5 +72,8 @@ func InspectorClient(inspectorEndpoint string, auth AuthConfig) (client *gopherc
 	default:
 		err = fmt.Errorf("Unknown auth type %s", auth.Type)
 	}
-	return
+	if err != nil {
+		return
+	}
+	return updateHTTPClient(client, caFile)
```

Comment:
I really like that this is orthogonal to whether we're using basic auth or not :+1: